lib/matplotlib/ticker.py

```diff
@@ -1323,23 +1333,12 @@ def view_limits(self, dmin, dmax):
 
 
 def scale_range(vmin, vmax, n=1, threshold=100):
-    dv = abs(vmax - vmin)
-    if dv == 0:     # maxabsv == 0 is a special case of this.
-        return 1.0, 0.0
-        # Note: this should never occur because
-        # vmin, vmax should have been checked by nonsingular(),
-        # and spread apart if necessary.
-    meanv = 0.5 * (vmax + vmin)
-    if abs(meanv) / dv < threshold:
-        offset = 0
-    elif meanv > 0:
-        ex = divmod(math.log10(meanv), 1)[0]
-        offset = 10 ** ex
-    else:
-        ex = divmod(math.log10(-meanv), 1)[0]
-        offset = -10 ** ex
-    ex = divmod(math.log10(dv / n), 1)[0]
-    scale = 10 ** ex
+    dv = abs(vmax - vmin)  # > 0 as nonsingular is called before.
+    meanv = (vmax + vmin) / 2
+    offset = (math.copysign(10 ** (math.log10(abs(meanv)) // 1), meanv)
```

Comment:
My style preference would be to use a conventional if/else construct here.  It can still be 3 lines:
```python
offset = 0
if abs(meanv) / dv >= threshold:
    offset = (the expression you have)
```